lib/scimitar_schema_extension.rb

```diff
@@ -28,11 +28,43 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
+class OpenProjectNameSchema < Scimitar::Schema::Base
+  def self.scim_attributes
+    @scim_attributes ||= [
+      Scimitar::Schema::Attribute.new(name: "familyName", caseExact: true, type: "string", required: true),
+      Scimitar::Schema::Attribute.new(name: "givenName",  caseExact: true, type: "string", required: true)
+    ]
+  end
+end
+
+class OpenProjectNameComplexType < Scimitar::ComplexTypes::Base
+  set_schema OpenProjectNameSchema
+end
+
 module ScimitarSchemaExtension
-  def scim_attributes
-    super + [Scimitar::Schema::Attribute.new(name: "externalId",
-                                             type: "string",
-                                             caseExact: true,
-                                             required: true)]
+  module Group
+    def scim_attributes
+      [
+        Scimitar::Schema::Attribute.new(name: "displayName", caseExact: true, type: "string", required: true),
+        Scimitar::Schema::Attribute.new(name: "members", multiValued: true, complexType: Scimitar::ComplexTypes::ReferenceMember,
+                                        mutability: "readWrite"),
+        Scimitar::Schema::Attribute.new(name: "externalId", type: "string", caseExact: true, required: true)
+      ]
+    end
+  end
+
+  module User
+    def scim_attributes
+      [
+        Scimitar::Schema::Attribute.new(name: "userName", caseExact: true, type: "string", uniqueness: "server", required: true),
+        Scimitar::Schema::Attribute.new(name: "name", caseExact: true, complexType: OpenProjectNameComplexType, required: true),
```

Comment:
is `caseExact` required here, since this is a complex type that consists of pieces that are case exact or not. I understand that `required` is needed both here and in the complex type, because we are neither allowed to skip the complex type nor its members.

Reply:
Yes, you understood correctly. Let's have it in both places.